crates/core/app/tests/app_can_propose_community_pool_spends.rs

```diff
@@ -0,0 +1,296 @@
+            },
+        }
+    };
+    plan.populate_detection_data(OsRng, 0);
+    let tx = client.witness_auth_build(&plan).await?;
+
+    // Execute the transaction, applying it to the chain state.
+    test_node
+        .block()
+        .with_data(vec![tx.encode_to_vec()])
+        .execute()
+        .instrument(error_span!("executing block with community pool deposit"))
+        .await?;
+    let post_deposit_snapshot = storage.latest_snapshot();
+    client.sync_to_latest(post_deposit_snapshot).await?;
+
+    // Now, make a governance proposal that we should spend community pool funds, to return
+    // the note back to the test wallet.
+    let mut plan = {
+        // XXX The submission should include a proposed transaction plan to spend pool funds.
+        // TODO: per the spec, the asset id and address are specified in base64. fix that!
+        // let value = note.value();
+        // let proposed_tx_plan = TransactionPlan {
+        //     actions: vec![
+        //         CommunityPoolSpend { value }.into(),
+        //         CommunityPoolOutput {
+        //             value,
+        //             address: *test_keys::ADDRESS_0,
+        //         }
+        //         .into(),
+        //     ],
+        //     memo: None,
+        //     detection_data: None,
+        //     transaction_parameters: TransactionParameters {
+        //         chain_id: TestNode::<()>::CHAIN_ID.to_string(),
+        //         ..Default::default()
+        //     },
+        // };
+        let proposal = ActionPlan::ProposalSubmit(ProposalSubmit {
+            proposal: Proposal {
+                id: 1_u64,
+                title: "return test deposit".to_owned(),
+                description: "a proposal to return the community pool deposit".to_owned(),
+                payload: penumbra_governance::ProposalPayload::CommunityPoolSpend {
+                    // XXX transaction_plan: proposed_tx_plan.encode_to_vec(),
+                    transaction_plan: TransactionPlan::default().encode_to_vec(),
+                },
+            },
+            deposit_amount: 0_u32.into(),
+        });
+        TransactionPlan {
+            actions: vec![proposal],
+            memo: None,
+            detection_data: None,
+            transaction_parameters: TransactionParameters {
+                chain_id: TestNode::<()>::CHAIN_ID.to_string(),
+                ..Default::default()
+            },
+        }
+    };
+    plan.populate_detection_data(OsRng, 0);
+    let tx = client.witness_auth_build(&plan).await?;
+
+    // Execute the transaction, applying it to the chain state.
+    test_node
+        .block()
+        .with_data(vec![tx.encode_to_vec()])
+        .execute()
+        .instrument(error_span!("executing block with governance proposal"))
+        .await?;
+    let post_proposal_snapshot = storage.latest_snapshot();
+    panic!("stop here"); // XXX this transaction does not execute successfuly, panic here for now.
```

Comment:
**question:** this currently fails with:

```
  2024-04-16T19:48:50.055373Z  INFO penumbra_app::server::consensus: deliver_tx failed, e: failed to deliver transaction

Caused by:
    0: check_stateless failed
    1: binding signature failed to verify
    2: Invalid signature.
    at crates/core/app/src/server/consensus.rs:207
    in penumbra_mock_consensus::block::execute with height: 2, time: 1713296930
    in app_can_propose_community_pool_spends::executing block with governance proposal
```

i believe this is because i should also be https://protocol.penumbra.zone/main/governance.html#proposal-nfts generating the `proposal_1_submitted` nft as an output(?) alongside the governance proposal itself.